ui/Modal/ProjectCreation.svelte

```diff
@@ -28,131 +33,30 @@
   } from "../DesignSystem/Component";
 
   let currentSelection;
+  let nameInput;
   export let content;
 
-  const projectNameMatch = "^[a-z0-9][a-z0-9._-]+$";
   const dispatch = createEventDispatcher();
 
   $: isNew = currentSelection === RepoType.New;
```

Comment:
I don't love having these two reactive variables - isn't a project always either `New` or `Existing`? I think this would make more sense with just one `isNew` variable:

```
 isNew ? { do something because it's new } : { do something else because it exists already }
```

It's confusing to have certain ui elements depend on `isExisting` and others on `isNew`.

Reply:
`currentSelection` is `null` at the beginning of the modal so that neither `isNew` and `isExisting` are `true`.
For some elements with the current set up, we need `isExisting`, like the tooltip and disabled-ness of the project name input field should only happen if the `currentSelection` is set and it is set to  `RepoType.Existing`. I guess it could be replaced by `currentSelection && currentSelection === RepoType.Existing`, but I find `isExisting` easier to read.